File: WarehouseWebsite.Domain/Models/Customers/Customer.cs
using System.ComponentModel.DataAnnotations;
using WarehouseWebsite.Domain.Models.Orders;

namespace WarehouseWebsite.Domain.Models.Customers
{
    public class Customer : BaseEntity
    {
        [Required]
        [StringLength(50, MinimumLength = 1)]
        public string Name { get; set; } = null!;

        [Required]
        [StringLength(50, MinimumLength = 1)]
        public string Surname { get; set; } = null!;

        [StringLength(100)]
        public string? Address { get; set; }

        [EmailAddress]
        public string? Email { get; set; }

        public IEnumerable<Order>? Orders { get; set; }
    }
}

File: WarehouseWebsite.Frontend/src/app/shopping/purchase/customer.ts
export class Customer {
  email: string;
  name: string;
  surname: string;
  address: string;

  constructor(email: string, name: string, surname: string, address: string) {
    this.email = email;
    this.name = name;
    this.surname = surname;
    this.address = address;
  }
}
